src/main/java/hudson/plugins/brakeman/BrakemanPublisher.java

```diff
@@ -91,65 +93,63 @@ public Action getProjectAction(final AbstractProject<?, ?> project) {
 
 	/** {@inheritDoc} */
 	@Override
-		public BuildResult perform(final Run<?, ?> build, final FilePath workspace, final PluginLogger logger) throws InterruptedException, IOException {
-			return publishReport(build, workspace);
-		}
+	public BuildResult perform(final Run<?, ?> build, final FilePath workspace, final PluginLogger logger) throws InterruptedException, IOException {
+		return publishReport(build, workspace, logger);
+	}
 
-		public BuildResult publishReport(final Run<?, ?> build, final FilePath workspace) throws InterruptedException, IOException {
-			FilePath brakemanOutput = new FilePath(workspace, this.outputFile);
+	public BuildResult publishReport(final Run<?, ?> build, final FilePath workspace, final PluginLogger logger) throws InterruptedException, IOException {
+		FilePath brakemanOutput = new FilePath(workspace, this.outputFile);
+		String output = brakemanOutput.readToString();
 
-			String output = brakemanOutput.readToString();
+		ParserResult project = new ParserResult(workspace);
 
-			ParserResult project = new ParserResult(workspace);
-			this.scan(output, project);
+		AbstractBrakemanScanner scanner;
+		String scannerType = getFileType(output);
+		if(scannerType == "JSON") {
+			scanner = new BrakemanJSONScanner();
+		} else {
+			scanner = new BrakemanTabsScanner();
+		}
+		boolean successfulScan = scanner.scan(output, project, logger);
+		BrakemanResult result = new BrakemanResult(build, getDefaultEncoding(), project, usePreviousBuildAsReference(), useOnlyStableBuildsAsReference());
+		if(successfulScan == false) {
+			logger.log(String.format("%s", "Oops, we encountered a problem when reading your reports file. Please make sure your Brakeman Output file is valid."));
+			result.setReason("Reading of Brakeman output failed. Check logs for more information.");
+			result.setResult(Result.FAILURE);
```

Comment:
~~hey, is setting Result.FAILURE on the BrakemanResult object sufficient to basically stop jenkins from processing the `BrakemanResults` to determine if the build failed or not or will jenkins still call `BrakemanResult.evaluateStatus()`?~~ looks like it is ... called `setResult()` with a FAILURE will mark the result as unsuccessful.